include/boost/math/special_functions/prime_sieve.hpp

```diff
@@ -0,0 +1,425 @@
+                std::thread t1([&small_primes] {
+                    boost::math::detail::prime_table(static_cast<Integer>(104729), small_primes);
+                });
+                std::thread t2([limit, &pre_generated_primes] {
+                    boost::math::detail::segmented_sieve(static_cast<Integer>(104729), limit, pre_generated_primes);
+                });
+
+                t1.join();
+                t2.join();
+                pre_generated_primes.insert(pre_generated_primes.begin(), small_primes.begin(), small_primes.end());
+            }
+
+            std::vector<std::thread> thread_manager {};
+            std::vector<std::vector<Integer>> prime_vectors(processor_count);
+            const Integer range_per_thread = upper_bound / (processor_count);
+            Integer current_lower_bound {limit + 1};
+            Integer current_upper_bound {current_lower_bound + range_per_thread};
+            Integer primes_in_range {static_cast<Integer>(current_upper_bound / std::log(static_cast<double>(current_upper_bound)) -
+                               current_lower_bound / std::log(static_cast<double>(current_lower_bound)))};
+
+            for(size_t i{}; i < processor_count - 1; ++i)
+            {
+                prime_vectors[i].reserve(primes_in_range);
+
+                std::thread t([current_lower_bound, current_upper_bound, &prime_vectors, i, &pre_generated_primes] {
+                    boost::math::detail::segmented_sieve(current_lower_bound, current_upper_bound, pre_generated_primes,
+                                                         prime_vectors[i]);
+                });
+
+                thread_manager.push_back(std::move(t));
+
+                current_lower_bound = current_upper_bound;
+                current_upper_bound += range_per_thread;
+                primes_in_range = static_cast<Integer>(current_upper_bound / std::log(static_cast<double>(current_upper_bound)) -
+                                                 current_lower_bound / std::log(static_cast<double>(current_lower_bound)));
+            }
+
+            prime_vectors.back().reserve(primes_in_range);
+
+            std::thread t([current_lower_bound, upper_bound, &prime_vectors, &pre_generated_primes] {
+                boost::math::detail::segmented_sieve(current_lower_bound, upper_bound, pre_generated_primes, prime_vectors.back());
+            });
+            thread_manager.push_back(std::move(t));
+
+            for(auto &thread : thread_manager)
+            {
+                thread.join();
+            }
+
+            primes.insert(primes.begin(), pre_generated_primes.begin(), pre_generated_primes.end());
+            for(auto &v : prime_vectors)
+            {
+                primes.insert(primes.end(), v.begin(), v.end());
+            }
+        }
+    }
+}
+
+template<class Integer, class Container>
+void prime_sieve(Integer upper_bound, Container &primes)
+{
+    prime_sieve(std::execution::par, upper_bound, primes);
+}
```

Comment:
Somewhat sadly you might say, but the standard algorithms default to a sequenced policy of execution.
I personally would err on the side of consistency and use the same default, but otherwise, at least Boost.Math should have a consistent default within the library. @jzmaddock ?